src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java

```diff
@@ -226,36 +230,66 @@ public void amendHearing(CaseData caseData, String caseTypeId) {
         }
     }
 
+    public void midEventAmendHearing(CaseData caseData, List<String> errors) {
+        if (!CollectionUtils.isEmpty(caseData.getHearingCollection())) {
+            for (HearingTypeItem hearingTypeItem : caseData.getHearingCollection()) {
+                if (!CollectionUtils.isEmpty(hearingTypeItem.getValue().getHearingDateCollection())) {
+                    for (DateListedTypeItem dateListedTypeItem
+                            : hearingTypeItem.getValue().getHearingDateCollection()) {
+                        addError(dateListedTypeItem, errors);
+                    }
+                }
+            }
+        }
+    }
+
+    private void addError(DateListedTypeItem dateListedTypeItem, List<String> errors) {
```

Comment:
Method name is generic and doesn't clearly communicate what it does

Reply:
changed method name